docs/docs/grpc.md

```diff
@@ -18,12 +18,134 @@ main = runGRpcApp 8080 "helloworld" quickstartServer
+
+We need to define a new record type (hence the name) that declares the services our client is going to consume. Remember that the names of the record fields **must match** exactly the methods in the service:
+
+```haskell
+import GHC.Generics (Generic)
+import Mu.GRpc.Client.Record
+
+data Call = Call
+  { getPerson :: MPersonRequest -> IO (GRpcReply MPerson)
+  , newPerson :: MPerson -> IO (GRpcReply MPersonRequest)
+  , allPeople :: IO (ConduitT () (GRpcReply MPerson) IO ())
+  } deriving Generic
+```
+
+Note that we had to derive `Generic`. We also need to tweak a little bit our `main` function:
+
+```diff
+main :: IO ()
+main = do
+   let config = grpcClientConfigSimple "127.0.0.1" 1234 False
+-  Right client <- setupGrpcClient' config
++  Right grpcClient <- setupGrpcClient' config
++  let client = buildService @Service @"" grpcClient
+   args <- getArgs
+```
+
+Instead of building our client directly, we can use `TypeApplications` to call `buildService` to create the actual gRPC client. (You can also use `ScopedTypeVariables` and something like `(Proxy :: Proxy Service)` but it's a bit more verbose).
+
+After that, let's have a look at an example implementation of the three service calls:
+
+```haskell
+import Text.Read (readMaybe)
+
+get :: Call -> String -> IO ()
+get client idPerson = do
+  let req = MPersonRequest $ readMaybe idPerson
+  putStrLn $ "GET: is there some person with id: " ++ idPerson ++ "?"
+  res <- getPerson client req
+  putStrLn $ "GET: response was: " ++ show res
+```
 
-### Using `TypeApplications`
+Notice the use of `readMaybe` to convert the strings to the appropiate type in a safe manner! 👆🏼
+
+```haskell
+add :: Call -> String -> String -> IO ()
+add client nm ag = do
+  let p = MPerson Nothing (Just $ T.pack nm) (readMaybe ag)
+  putStrLn $ "ADD: creating new person " ++ nm ++ " with age " ++ ag
+  res <- newPerson client p
+  putStrLn $ "ADD: was creating successful? " ++ show res
+
+watching :: Call -> IO ()
+watching client = do
+  replies <- allPeople client
+  runConduit $ replies .| C.mapM_ print
+```
+
+### Using `TypeApplications`
+
+With `TypeApplications` none of the above is needed, all you need to do is call `gRpcCall` with the appropiate service name as a type-level string, and the rest just _magically_ works! ✨
+
+```haskell
+import Mu.GRpc.Client.TyApps
+
+get :: GrpcClient -> String -> IO ()
+get client idPerson = do
+  let req = MPersonRequest $ readMaybe idPerson
+  putStrLn $ "GET: is there some person with id: " ++ idPerson ++ "?"
+  response :: GRpcReply MPerson
+    <- gRpcCall @Service @"getPerson" client req
+  putStrLn $ "GET: response was: " ++ show response
+```
+
+Notice that the type signatures of our functions needed to change to receive the `GrpcClient` as an argument, instead of our custom record type.
+
+```haskell
+add :: GrpcClient -> String -> String -> IO ()
```

Comment:
Could we factor out the implementation of `add` above, next to the `main` function, and make it parametric on a `Person -> IO String` function? Same for the `get` function.